app-main/public/js/index.js

```diff
@@ -43,6 +47,38 @@ modalLessonClose.addEventListener("click", handleCloseLessonModal);
 clearBtn.addEventListener("click", handleClear);
 overlay.addEventListener("click", handleCloseLessonModal);
 formElement.addEventListener("keyup", handleClearBtn);
+addTagButton.addEventListener("click", handleTagVisibility);
+tagSelectors.addEventListener("click", handleTagSelect);
+
+// create tag colors
+TAGS.forEach((tag) => {
```

Comment:
I like that suggestion, though one problem I worry about is that styling based off of hex values defined in a class might be messy if we want to change the colors of the tags. Is there a way for CSS to grab the hex values after selecting each tag?

Reply:
Good point regarding the hex values. Thinking about this now... a better approach would be to have these tags as a list of text id's or classes or even attributes (so we can access them with a CSS selector), for instance:

```
const tags = ['red', 'yellow', 'blue'];

// or

const tags = ['programming', 'in progress', 'frameworks'];
```
we can then handle the colour's in the stylesheet. We could use a CSS variable to define these, for example:

```
--in-progress: #f7b501;

// example selector
.in-progress {
  background-color: var(--in-progress);
}
```

That way, we only have to set our hex values in one place. Using default named tags such as `in-progress`, `frameworks` etc may be preferable so that we can decouple the name from the colour.

Reply:
I love your suggestion! I changed the tags to human readable default names and used CSS to style the background colors. I also kept the tags as classes to be thrown onto the divs, because in the future we may want to style other things with consistent colors, and so all we'd need to do is throw the class (ie in-progress) on the element.